### PR TITLE
[kube-prometheus-stack] Inhibit info alerts unless other alerts fire

### PR DESCRIPTION
**Pull request by wiegandf**
#### What this PR does / why we need it:
Sync of https://github.com/prometheus-operator/kube-prometheus/pull/1507

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/prometheus/rules-1.14/general.rules.yaml [hunk 1]
```diff
@@ -61,4 +61,29 @@ spec:
 {{ toYaml .Values.defaultRules.additionalRuleLabels | indent 8 }}
 {{- end }}
 {{- end }}
+{{- if not (.Values.defaultRules.disabled.InfoInhibitor | default false) }}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus/rules-1.14/general.rules.yaml:64]: https://github.com/prometheus-community/helm-charts/blob/fb2f8597cc2539c9ec9ef3edab1e4f5a04863b19/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/general.rules.yaml#L2-L4

File: charts/kube-prometheus-stack/templates/prometheus/rules-1.14/k8s.rules.yaml [hunk 2]
```diff
@@ -160,4 +160,14 @@ spec:
       labels:
         workload_type: statefulset
       record: namespace_workload_pod:kube_pod_owner:relabel
+    - expr: |-
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus/rules-1.14/k8s.rules.yaml:163]: https://github.com/prometheus-community/helm-charts/blob/fb2f8597cc2539c9ec9ef3edab1e4f5a04863b19/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/k8s.rules.yaml#L2-L4

File: charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-apps.yaml [hunk 3]
```diff
@@ -188,7 +188,7 @@ spec:
              !=
             0
           ) or (
-            kube_daemonset_updated_number_scheduled{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
+            kube_daemonset_status_updated_number_scheduled{job="kube-state-metrics", namespace=~"{{ $targetNamespace }}"}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-apps.yaml:191]: https://github.com/prometheus-community/helm-charts/blob/fb2f8597cc2539c9ec9ef3edab1e4f5a04863b19/charts/kube-prometheus-stack/templates/prometheus/rules-1.14/kubernetes-apps.yaml#L2-L4

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by monotek** [comment]
> @monotek sorry for not following the process, I thought I would.
> 
> I just did the last two points https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/hack/README.md?plain=1#L20-L21
> 
> The alerts I want to sync are already on `prometheus-operator/kube-prometheus` master, so what am I missing?

You have to adjust the python scipt or your changes will be overwritten on next sync.

**Comment by wiegandf** [comment]
I am sorry I don't get what needs to be changed. The readme doesn't contain any instructions for it and the python script doesn't links to main branch. https://github.com/prometheus-community/helm-charts/blob/3663f3c71b22938891318a352fea40208261e0de/charts/kube-prometheus-stack/hack/sync_prometheus_rules.py#L33

**Pull request closed without merging**